src/sentry/integrations/slack/actions/notification.py

```diff
@@ -117,12 +127,81 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
+                )
+
+            rule_fire_history_id = None
+            try:
+                rule_fire_history = RuleFireHistory.objects.get(notification_uuid=notification_uuid)
+                rule_fire_history_id = rule_fire_history.id
+                if not rule_id:
+                    rule_id = rule_fire_history.rule.id
+            except Exception as err:
+                _default_logger.info(
+                    "integration.slack.actions: Failed to get rule fire history", exc_info=err
+                )
+
+            new_notification_message_object = NewIssueAlertNotificationMessage(
+                rule_fire_history_id=rule_fire_history_id, rule_action_uuid=rule_action_uuid
+            )
+
+            # Only try to get the parent notification message if the organization is in the FF
+            # We need to search by rule action uuid and rule id, so only search if they exist
+            if (
+                features.has(
+                    "organizations:slack-thread-issue-alert", event.group.project.organization
+                )
+                and rule_action_uuid
+                and rule_id
+            ):
+                try:
+                    parent_notification_message = self._repository.get_parent_notification_message(
+                        rule_id=rule_id,
+                        group_id=event.group.id,
+                        rule_action_uuid=rule_action_uuid,
+                    )
+                except Exception:
+                    # if there's an error trying to grab a parent notification, don't let that error block this flow
+                    # we already log at the repository layer, no need to log again here
+                    pass
+                else:
+                    if parent_notification_message:
+                        # If a parent notification exists for this rule and action, then we can reply in a thread
+                        # Make sure we track that this reply will be in relation to the parent row
+                        new_notification_message_object.parent_notification_message_id = (
+                            parent_notification_message.id
+                        )
+                        # To reply to a thread, use the specific key in the payload as referenced by the docs
+                        # https://api.slack.com/methods/chat.postMessage#arg_thread_ts
+                        payload["thread_ts"] = parent_notification_message.message_identifier
+
             client = SlackClient(integration_id=integration.id)
             try:
-                client.post(
+                response = client.post(
                     "/chat.postMessage", data=payload, timeout=5, log_response_with_error=True
                 )
             except ApiError as e:
+                # Record the error code and details from the exception
+                new_notification_message_object.error_code = e.code
+                new_notification_message_object.error_details = {
```

Comment:
this is pretty cool :O

Reply:
yeah, we're using this to determine broken rule actions actually haha